core/src/main/java/org/axonframework/eventhandling/saga/repository/AssociationValueMap.java

```diff
@@ -198,5 +198,52 @@ public int compare(SagaAssociationValue o1, SagaAssociationValue o2) {
             }
             return value;
         }
+
+        /**
+         * Copied from Spring's ObjectUtils because Spring dependency is no longer allowed and I
+         * did not want to introduce any regressions by handcrafting something worse.
+         */
+        protected static boolean nullSafeEquals(Object o1, Object o2) {
```

Comment:
right so that's what you meant, i will have a look tonight to use that one instead.